iroh-gateway/src/core.rs

```diff
@@ -298,87 +426,167 @@ mod tests {
-                .unwrap();
-            let req = hyper::Request::builder()
-                .method("HEAD")
-                .uri(uri)
-                .body(hyper::Body::empty())
-                .unwrap();
-            client.request(req).await.unwrap()
-        };
+        assert_eq!(http::StatusCode::MOVED_PERMANENTLY, res.status());
+        assert_eq!(
+            format!(
+                "http://{}.ipfs.localhost:{}/",
+                test_setup.root_cid,
+                test_setup.gateway_addr.port()
+            ),
+            res.headers().get("Location").unwrap().to_str().unwrap()
+        );
 
-        assert_eq!(http::StatusCode::OK, res.status());
-        assert!(res.headers().get("content-length").is_some());
-        assert_eq!(res.headers().get("content-length").unwrap(), "5");
+        // IPNS
+        let res = do_request(
+            "GET",
+            &format!("localhost:{}", test_setup.gateway_addr.port()),
+            "/ipns/k51qzi5uqu5dlvj2baxnqndepeb86cbk3ng7n3i46uzyxzyqj2xjonzllnv0v8",
+            None,
+        )
+        .await;
 
-        let (body, _) = res.into_body().into_future().await;
-        assert!(body.is_none());
+        assert_eq!(http::StatusCode::MOVED_PERMANENTLY, res.status());
+        assert_eq!(
+            format!(
+                "http://k51qzi5uqu5dlvj2baxnqndepeb86cbk3ng7n3i46uzyxzyqj2xjonzllnv0v8.ipns.localhost:{}/",
+                test_setup.gateway_addr.port()
+            ),
+            res.headers().get("Location").unwrap().to_str().unwrap()
+        );
 
-        core_task.abort();
-        core_task.await.unwrap_err();
-        store_task.abort();
-        store_task.await.unwrap_err();
+        // Test that IPNS records are recoded to base36
+        let res = do_request(
+            "GET",
+            &format!("localhost:{}", test_setup.gateway_addr.port()),
+            "/ipns/bafyreihyrpefhacm6kkp4ql6j6udakdit7g3dmkzfriqfykhjw6cad5lrm",
+            None,
+        )
+        .await;
+
+        assert_eq!(http::StatusCode::MOVED_PERMANENTLY, res.status());
+        assert_eq!(
+            format!(
+                "http://k2jvslbl2n1p4suo7yr973y3v7pfautpxba2jeb9fpmjil0l3lppcgi3.ipns.localhost:{}/",
+                test_setup.gateway_addr.port()
+            ),
+            res.headers().get("Location").unwrap().to_str().unwrap()
+        );
+
+        // IPNS + DNSLink
+        let res = do_request(
+            "GET",
+            &format!("localhost:{}", test_setup.gateway_addr.port()),
+            "/ipns/ipfs-subdomain.ipfs-domain.eth",
+            None,
+        )
+        .await;
+
+        assert_eq!(http::StatusCode::MOVED_PERMANENTLY, res.status());
+        assert_eq!(
+            format!(
+                "http://ipfs--subdomain-ipfs--domain-eth.ipns.localhost:{}/",
+                test_setup.gateway_addr.port()
+            ),
+            res.headers().get("Location").unwrap().to_str().unwrap()
+        );
+
+        // X-Forwarded-Proto
```

Comment:
nit: missing the same test for `x-forwarded-host`

Reply:
Done